ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -501,7 +502,11 @@ protected Boolean findNextCompactionAndExecute(boolean collectGenericStats, bool
       }
     } catch (TException | IOException t) {
       LOG.error("Caught an exception in the main loop of compactor worker " + workerName, t);
-      markFailed(ci, t.getMessage());
+
+      if (ci != null) {
```

Comment:
I think it's better to move NULL check inside of markFailed, it's already there but joined with extra condition

Reply:
I moved it here with additional logging. It has the exact same pattern as if the MetaStoreClient is null:
```java
if (ci == null) {
      LOG.warn("CompactionInfo client was null. Could not mark failed: {}", ci);
      return;
    }
```